sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/pipeline/_pipeline_expression.py

```diff
@@ -334,17 +334,17 @@ def _handle_component_output(
                 _seen_input = _expression_inputs[_name]
                 if isinstance(_seen_input.value, PipelineInput):
                     if not _has_prefix:
-                        _name = f"{_component_output._owner.component.name}__{_component_output._name}"
+                        _name = f"{_component_output._owner.component.name}_{_component_output._name}"
```

Comment:
why this?

Reply:
offline synced on this, the root cause for this change is in some cases, the `command` for mldesigner command component is not expected for current code, need to align this and have some methods to sync this contract.

Reply:
offline synced

Reply:
I have checked related logic for mldesigner and find that I was wrong on the understanding of mldesigner component `command`: I used to think there is some special logic for mldesigner, but in fact there isn't and it is service-side execute related overwrite.
So here we can use something like `${{inputs.xxx}}/${{outputs.output}}` in `command` and everything will work fine under v2 contract.